test/cases/ostree-ami-image.sh

```diff
@@ -0,0 +1,1105 @@
+SG_OUTPUT=sec_group.json
+aws ec2 create-security-group --group-name mysecuritygroup --description "edge-ami security group" --vpc-id "${VPC_ID}" | tee "${SG_OUTPUT}"
+SEC_GROUP_ID=$(cat "${SG_OUTPUT}" | jq -r '.GroupId')
+# Allow inbound ssh connections
+aws ec2 authorize-security-group-ingress --group-id "${SEC_GROUP_ID}" --protocol tcp --port 22 --cidr 0.0.0.0/0
+
+# Create instance market options
+MARKET_OPTIONS=spot-options.json
+tee "${MARKET_OPTIONS}" > /dev/null << EOF
+{
+  "MarketType": "spot",
+  "SpotOptions": {
+    "MaxPrice": "0.1",
+    "SpotInstanceType": "one-time",
+    "InstanceInterruptionBehavior": "terminate"
+  }
+}
+EOF
+
+##################################################################
+##
+## Install and test EC2 instance with edge-ami (BIOS)
+##
+##################################################################
+# Create AMI image from EBS snapshot
+greenprint "Register AMI, create image from snapshot."
+REGISTERED_AMI_NAME=reg_rhel_edge_ami
+REGISTERED_AMI_ID=output_ami_id.json
+SNAPSHOT_ID=$(cat "${IMPORT_SNAPSHOT_TASK}" | jq -r '.ImportSnapshotTasks[].SnapshotTaskDetail.SnapshotId')
+aws ec2 register-image \
+    --name "${REGISTERED_AMI_NAME}" \
+    --root-device-name /dev/xvda \
+    --architecture "${ARCH}" \
+    --ena-support \
+    --sriov-net-support simple \
+    --virtualization-type hvm \
+    --block-device-mappings DeviceName=/dev/xvda,Ebs={SnapshotId="${SNAPSHOT_ID}"} DeviceName=/dev/xvdf,Ebs={VolumeSize=10} \
+    --boot-mode legacy-bios \
+    --output json > "${REGISTERED_AMI_ID}"
+
+AMI_ID=$(cat "${REGISTERED_AMI_ID}" | jq -r '.ImageId')
+tag_ec2_ami "${AMI_ID}"
+# TO DO: Remove describe tags functions
+tag_describe_resource "${AMI_ID}"
+
+# Launch Instance
+greenprint "💻 Launch instance from AMI"
+INSTANCE_OUT_INFO=instance_output_info.json
+aws ec2 run-instances \
+    --image-id "${AMI_ID}" \
+    --count 1 \
+    --instance-type "${INSTANCE_TYPE}" \
+    --instance-market-options file://"${MARKET_OPTIONS}" \
+    --key-name "${AMI_KEY_NAME}" \
+    --security-group-ids "${SEC_GROUP_ID}" \
+    --subnet-id "${SUBN_ID}" \
+    --associate-public-ip-address > "${INSTANCE_OUT_INFO}"
+cat "${INSTANCE_OUT_INFO}"
+
+# wait for instance running
+sleep 5
+
+# get instance public ip
+INSTANCE_ID=$(jq -r '.Instances[].InstanceId' "${INSTANCE_OUT_INFO}")
+tag_ec2_instance "${INSTANCE_ID}"
+# TO DO: Remove describe tags functions
+tag_describe_resource "${INSTANCE_ID}"
+BIOS_GUEST_ADDRESS=$(aws ec2 describe-instances --instance-ids "${INSTANCE_ID}" --query 'Reservations[*].Instances[*].PublicIpAddress' --output text)
+
+# Check for ssh ready to go.
+greenprint "🛃 Checking for SSH is ready to go"
+for LOOP_COUNTER in $(seq 0 30); do
+    RESULTS="$(wait_for_ssh_up $BIOS_GUEST_ADDRESS)"
+    if [[ $RESULTS == 1 ]]; then
+        echo "SSH is ready now! 🥳"
+        break
+    fi
+    sleep 10
+done
+
+# With new ostree-libs-2022.6-3, edge vm needs to reboot twice to make the /sysroot readonly
```

Comment:
I think the ami image doesn't need reboot again, you can double confirm with devel.